jsonnet_test.go

```diff
@@ -395,3 +396,74 @@ func TestGlobalBinding(t *testing.T) {
 		t.Errorf("output is not correct, want '%s' got '%s'", expected, actual)
 	}
 }
+
+func TestNotifier_OnGeneratedValue(t *testing.T) {
+	notifier := &testNotifier{}
+	vm := MakeVM()
+	vm.Notifier(notifier)
+	vm.NativeFunction(&NativeFunction{
+		Name:   `decorate`,
+		Params: ast.Identifiers{"str"},
+		Func: func(params []interface{}) (interface{}, error) {
+			return fmt.Sprintf("~%s~", params[0].(string)), nil
+		},
+	})
+
+	// Jsonnet code
+	code := `
+local Person(name='Alice') = {
+  name: if true then std.native('decorate')(name) else null,
+};
+local Do() = {
+  person1: Person(),
+  person2: Person('Bob'),
+  other: [Person('Foo'), Person('Bar')],
+};
+Do()
```

Comment:
Mame tu Jsonnet net kod, kde sa zlozito generuje hodnota, pomocou nativnej (tj. definovanej v Go kode) funkcie `decorate`.

Reply:
V KAC mame este jedno vylepsenie, ze sa namiesto `std.native('decorate')(..)`, moze volat `decorate(...)`, ale tu to nie je.